frontend/src/metabase/plugins/index.ts

```diff
@@ -68,8 +69,9 @@ export const PLUGIN_ADMIN_USER_MENU_ROUTES = [];
 // authentication providers
 export const PLUGIN_AUTH_PROVIDERS = [] as any;
 
-// Only show the password tab in account settings if these functions all return true
-export const PLUGIN_SHOW_CHANGE_PASSWORD_CONDITIONS = [];
+// Only show the password tab in account settings if these functions all return true.
+// Otherwise, the user is logged in via SSO and should hide first name, last name, and email field in profile settings metabase#23298.
+export const PLUGIN_IS_PASSWORD_USER: ((user: User) => boolean)[] = [];
```

Comment:
Rename this plugin so I could infer the opposite of these conditions as SSO users. e.g.
```ts
const isSsoUser = !PLUGIN_IS_PASSWORD_USER.every(predicate => predicate(user))
```